contracts/NodeStorage.sol

```diff
@@ -98,7 +101,33 @@ contract NodeStorage {
         }
     }
 
-    function calculateKey(string memory _enodeId, string memory _ip, uint16 _port) public pure returns(uint256) {
-        return uint256(keccak256(abi.encodePacked(_enodeId, _ip, _port)));
+    function upgradeHostValidation (bool activate) public onlyLatestVersion returns (bool) {
+        if (hostValidation == activate) {
+            return true;
+        }
+
+        // First we reset old map entries
+        enode memory entry;
+        for (uint256 index = 0; index < allowlist.length; index++) {
+            entry = allowlist[index];
+            indexOf[calculateKey(entry.enodeId, entry.ip, entry.port)] = 0;
+        }
+
+        hostValidation = activate;
+        
+        for (uint256 index = 0; index < allowlist.length; index++) {
+            entry = allowlist[index];
+            indexOf[calculateKey(entry.enodeId, entry.ip, entry.port)] = index + 1;
+        }
+
+        return true;
+    }
+
+
+    function calculateKey(string memory _enodeId, string memory _host, uint16 _port) public view returns(uint256) {
+        if (hostValidation) {
+            return uint256(keccak256(abi.encodePacked(_enodeId, _host, _port)));
+        }
+        return uint256(keccak256(abi.encodePacked(_enodeId, _port)));
```

Comment:
Do we want to flip this on its head? 

`validateWithJustEnode` 

When I think of host I normally think of port also?

Reply:
agree if we're ignoring the host it makes sense to ignore the port

Reply:
Updated with only enodeId validation for no-host.
Renamed function.